_posts/2024-05-03-trino-snowflake-bloomberg-oss-win.md

```diff
@@ -0,0 +1,294 @@
+also known as cloud data platform, that had no way to test this integration
+that was 
+[eagerly](https://github.com/trinodb/trino/pull/2551#issuecomment-873082280)
+[sought](https://github.com/trinodb/trino/issues/1863)
+[by many](https://github.com/trinodb/trino/issues/7247). After an
+[initial attempt](https://github.com/trinodb/trino/pull/2551) by my friend
+[Phillipe Gagnon](https://www.linkedin.com/in/pfgagnon), a similar pattern
+emerged [with the second pull request](https://github.com/trinodb/trino/pull/10387)
+where the development velocity started strong and after some months stagnated.
+
+### Cognitive surplus and communication deficit
+
+One of the most unfortunate repeated issues I see in the open source projects
+I've contributed to and helped maintain is that various well-known larger
+objectives known among the core group often move so fast, while individual
+updates that don't fit in a larger project narrative has higher likelihood to
+get lost in the shuffle. As an open source project grows, you end up with a
+cognitive surplus in the form of an abundance of bright people willing to share
+their time, intellect, and experience with a larger community. 
+
+Often both contributors and maintainers are so busy with their day jobs,
+families, and self care, that they dedicate most of their remaining energy to
+ensuring they write quality code and tests to the best of their ability. Lack of
+upfront communication to validate ideas from newer contributors, and lack of
+communication by maintainers who see a large number of issues to address are
+two communication issues that stagnate a project. Maintainers are often doers
+that see more value in addressing quick-win work that flows from the
+well-established contributors of the project. Followthrough on either side can
+be difficult as newcomers don't want to be rude and maintainers accidentally
+forget or hope someone else will take the time to address the issues on that
+pull request. 
+
+![](https://i.snap.as/kbwKpWa7.webp)
+
+Waiting for your work to be reviewed by someone in the community kind of works
+like a wishing well, you toss in a coin (i.e. your time and effort represented
+as code and a pull request) and hope your wish of getting your code reviewed
+and merged comes true. The satisfaction of hypothetical developer(s) that
+benefit from your small and significant change floods your mind and you feel
+like you’ve improved humanity just that one little bit more. 
+
+Maintainers are in a constant state of pulling triage on all the surplus of
+innovation being thrown at them and simultaneously trying to look for more help
+reviewing and being the expert at some areas of the code. As you can imagine,
+good communication can be hard to come by as many newcomers are strangers and
+concerned they are wasting precious time by asking too many questions rather
+than just showing a proof of concept. This backfires for multiple reasons that
+go outside of the scope of this post.
```

Comment:
I think you could briefly touch on them if you wanted, this feels a little abrupt...

Reply:
```
This backfires when developers will spend
a large portion of their time developing a solution that is not compatible with the
project, and maintainers will lose the opportunity to quickly spin up on the value
of the new feature. This is why regular
[contributor meetings](https://github.com/trinodb/trino/wiki/Contributor-meetings)
help solve both of these issues synchronously to cut out the delayed feedback loops.
```